Auth_phpbb.php

```diff
@@ -504,15 +590,50 @@ public function getCanonicalName( $username )
 
         // Query Database.
         $fresStatement = $fresMySQLConnection->prepare($fstrMySQLQuery);
-        $fresStatement->bind_param('s', $username);
+        $fresStatement->bind_param('s', $this->_phpBBUserName); // bind_param escapes the string
         $fresStatement->execute();
 
         // Bind result
-        $fresStatement->bind_result($resultUsernameClean);
+        $fresStatement->bind_result($resultWikiUsername);
 
         while($fresStatement->fetch())
         {
-            return ucfirst($resultUsernameClean);
+            $this->_wikiUserName = ucfirst($resultWikiUsername); // Preserve capped phpBB username when wikified version is valid
+            return $this->_wikiUserName;
+        }
+
+        // If here, username is invalid or is incompatible with wiki username.
```

Comment:
The intent here was that if the username is already valid, let's save on performance by quitting the script and not hitting the database again.  If there is a pointless profile username translation, it could be an erroneous admin entry to the wrong account, preventing an accidental hijack.  If it is merely an intentional alias, preserving a consistency of identity seems somehow preferable.  Or just create a second forum account.